reana_server/scheduler/consumer.py

```diff
@@ -173,19 +215,27 @@ def on_message(self, workflow_submission, message):
                     exc_info=True,
                 )
                 requeue = False
+                should_fail_workflow_if_not_started = False
```

Comment:
This is due to this comment - https://github.com/reanahub/reana-server/pull/411#issuecomment-952969077 . If the workflow is still running and it is fine but we cancel it in `scheduler`, the user will be confused on why workflow failed.

*note(non-blocking):* It would be much better if we can easily distinguish between error types - HTTPConflict doesn't say much. Instead of `HTTPConflict` error, would be good to get `DuplicatedStartError` and similar. Any ideas on that?

Reply:
I haven't tested this edge case for now, but I think it should be transparent to user and we don't need to `fail` the workflow because of it. In case RabbitMQ gets restarted and duplicate message is sent to consumer, the workflow should still continue running and a duplicate message should be dropped silently. WDYT? In this case we could get rid of `should_fail_workflow_if_not_started`

Reply:
That is why I added `should_fail_workflow_if_not_started=False`, so workflow **does not** fail in case of a duplicate message.

Maybe it is confusing to understand my intentions in the code. Do you have suggestions on how to make it more clear?